src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6322,12 +6358,182 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
+                : new InterpolatedStringHandlerContext(operation, outParameterFlowCapture, handlerCaptureId);
+
+            VisitAndCapture(operation.HandlerCreation, handlerCaptureId);
+
+            if (operation.HandlerCreationHasSuccessParameter)
+            {
+                // Branch on the success parameter to the next block
+                Debug.Assert(resultBlock != null);
+                Debug.Assert(outParameterPlaceholder != null);
+                Debug.Assert(outParameterFlowCapture != -1);
+
+                // if (!outParameterFlowCapture) goto resultBlock;
+                // else goto next block;
+                ConditionalBranch(new FlowCaptureReferenceOperation(outParameterFlowCapture, outParameterPlaceholder.Syntax, outParameterPlaceholder.Type, constantValue: null), jumpIfTrue: false, resultBlock);
+                _currentBasicBlock = null;
+            }
+
+            LeaveRegionsUpTo(resultRegion);
+            ImmutableArray<IInterpolatedStringAppendOperation> appendCalls = collectAppendCalls(operation);
+
+            int appendCallsLength = appendCalls.Length;
+            for (var i = 0; i < appendCallsLength; i++)
+            {
+                var appendRegion = new RegionBuilder(ControlFlowRegionKind.LocalLifetime);
+                EnterRegion(appendRegion);
+                var appendCall = appendCalls[i];
+                if (operation.HandlerAppendCallsReturnBool)
+                {
+                    Debug.Assert(resultBlock != null);
+
+                    IOperation visitedAppendCall = VisitRequired(appendCall.AppendCall);
+                    if (i == appendCallsLength - 1)
+                    {
+                        // No matter the result, we're going to the result block next. So just visit the statement, and if the current block can be
+                        // combined with the result block, the compaction machinery will take care of it
+                        AddStatement(visitedAppendCall);
+                        _currentBasicBlock = null;
+                    }
+                    else
+                    {
+                        // if (!appendCall()) goto result else goto next block
+                        ConditionalBranch(visitedAppendCall, jumpIfTrue: false, resultBlock);
+                        _currentBasicBlock = null;
+                    }
+                }
+                else
+                {
+                    VisitStatement(appendCall.AppendCall);
+
+                    if (i == appendCallsLength - 1 && operation.HandlerCreationHasSuccessParameter)
+                    {
+                        Debug.Assert(resultBlock != null);
+                        _currentBasicBlock = null;
+                    }
+                }
+                LeaveRegionsUpTo(appendRegion);
+                LeaveRegion();
```

Comment:
It looks like these two statements can be replaced with ```LeaveRegionsUpTo(resultRegion)```, which actually makes it clear where we are going to end up.